File: huawei/HJ5/HJ5.cpp
#include <iostream>

using namespace std;

// int main(){
//     string s;
//     while(cin>>s) {
//         cout << stoi(s, 0, 16) << endl;
//     }
//     return 0;
// }

int main() {
    string s;
    int res = 0, temp = 0;
    while(cin>>s) {
        for (int i = 2; i < s.size(); i++){
            if (s[i]>='0' && s[i]<='9'){
                temp = s[i] - '0';
            }else{
                temp = s[i] - 'A' + 10;
            }
            res = res*16 + temp;
        }
        cout << res << endl;
    }
    return 0;
}

File: huawei/HJ5/HJ5.py
# while True:
#     try:
#         s = input()
#         print(int(s, 16))
#     except:
#         break
dic = {'A':10, 'B':11, 'C':12, 'D':13, 'E':14, 'F':15, '0':0, '1':1, '2':2, '3':3, '4':4, '5':5, '6':6, '7':7, '8':8, '9':9}

while True:
    try:
        s = input()
        length = len(s)
        res = 0
        for i in range(2,length):
            temp = dic[s[i]]
            res = res * 16 + temp
        print(res)
    except:
        break